oeps/oep-0013.rst

```diff
@@ -0,0 +1,672 @@
+  service's deprecation policy.
+
+- *Must*: Additive changes to the contract of the API, or the results, should
+  not bump the major version, unless those changes are non-optional, and break
+  existing contracts with the API.
+
+Pagination
+----------
+
+We use the conventions established by `DRF
+<http://www.django-rest-framework.org/api-guide/pagination/#pagenumberpagination>`_,
+with some extra fields:
+
+- ``next``: A URL to the next page, or null.
+- ``previous``: A URL to the previous page, or null.
+- ``count``: The total number of items.
+- ``num_pages``: The total number of pages.
+- ``results``: List of results.
+
+The query parameters are as follows:
+
+- ``page`` - the page of results (zero indexed)
+- ``page_size`` - the number of results to return per page
+- ``sort_order`` - a string indicating the desired order (no direction
+  modifiers)
+
+Documentation
+-------------
+
+Work with the Docs team to update the `edX platform API documentation
+<http://edx-platform-api.readthedocs.org/en/latest/>`_.
+
+**Format**
+
+Use `Swagger <http://swagger.io/specification/>`_ to generate API documentation
+where possible.
+
+**Docstrings**
+
+Following are our guidelines for Django Rest Framework API implementations.
+
+edX API docstrings serve two purposes:
+
+- They are interpreted by DRF to provide dynamic API documentation
+
+    e.g. https://courses.edx.org/api/user/v1/accounts/AndyA
+
+    See the `DRF documentation
+    <http://tomchristie.github.io/rest-framework-2-docs/topics/documenting-your-api>`_
+    for more details.
+
+- They are compiled by the Sphinx Napoleon extension to be included in our
+  published API documentation set
+
+    http://edx-platform-api.readthedocs.org/en/latest/
+
+These two tools put constraints on the format of the docstring, so we developed
+the following compromise that should work for both:
+
+- Add the endpoint documentation as a docstring on your DRF view class
+
+    Note: individual method implementation docstrings are not currently used,
+    but are still worth including for readers of the code.
+
+- Use Markdown format, and do not include any HTML as it will render incorrectly
+  in one or both tools
+
+- Use double asterisks (bold) around headings
+
+- Be careful with the indentation as the Sphinx generation for ReadTheDocs is
+  very finicky
+
+- Provide at least the following sections:
+
+    Use Cases
+
+    Example Requests
+
+    Response Values for GET, Response Values for PUT etc if you implement
+    multiple methods)
+
+
+Here is a representative docstring (from
+https://github.com/edx/edx-platform/blob/master/common/djangoapps/enrollment/views.py#L67)::
+
+    class EnrollmentView(APIView, ApiKeyPermissionMixIn):
+    """
+    **Use Cases**
+    
+        Get the user's enrollment status for a course.
+    
+    **Example Requests**:
+    
+        GET /api/enrollment/v1/enrollment/{user_id},{course_id}
+    
+    **Response Values**
```

Comment:
This is not necessary when using Swagger. Also, docstrings like these tend to grow stale as APIs evolve since I don't necessarily have to touch a view to modify a serializer.